pkg/controller/grouping/group_entity_index.go

```diff
@@ -82,6 +84,12 @@ type Interface interface {
 	// DeleteExternalEntity deletes an ExternalEntity from the index. If any existing groups are affected, eventHandlers
 	// will be called with the affected groups.
 	DeleteExternalEntity(ee *v1alpha2.ExternalEntity)
+	// AddNode adds or updates a Node to the index. If any existing groups are affected, eventHandlers will be called with
+	// the affected groups.
+	AddNode(node *v1.Node)
```

Comment:
When NodeSelector is used in AddressGroup, the current code just calculate the selected Nodes via `c.nodeLister.List(selector)`, see https://github.com/antrea-io/antrea/blob/22d70be28066735c0bb5736ebaf89aac5df15591/pkg/controller/networkpolicy/networkpolicy_controller.go#L1137-L1140 
This is based on the assumption that the scale of Node is smaller than Pod, and Node events that we care about are much less frequent.
I think this PR should do the same to be consistent and to avoid making the grouping interface more complicated until it's necessary.